src/createQueryGroup.ts

```diff
@@ -19,41 +21,52 @@ export function createQueryGroup<
   type Group = Query<TAction, QueryOptions<TAction>>
 
   const intervalController = createIntervalController()
-
+  let lastExecuted: number | undefined = undefined
+  
   const response = ref<Group['response']>()
   const error = ref<Group['error']>()
   const errored = ref<Group['errored']>(false)
-  const lastExecuted = ref<number>()
   const executing = ref<Group['executing']>(false)
-  const executed = computed<Group['executed']>(() => lastExecuted.value !== undefined)
+  const executed = ref<Group['executed']>(false)
   const { promise, resolve } = Promise.withResolvers()
 
   const subscriptions = new Map<number, QueryOptions<TAction>>()
   const nextId = createSequence()
   const tags = new Set<QueryTagKey>()
 
-  async function execute(): Promise<void> {
+  async function execute(): Promise<AwaitedQuery<TAction>> {
+    lastExecuted = Date.now()
     executing.value = true
 
     try {
       const value = await action(...parameters)
       
       setResponse(value)
       setTags()
-      
-      error.value = undefined
-      errored.value = false
-    } catch(err) {
-      setError(err)
+      setNextExecution()
+
+      return response.value
+    } catch(error) {
+      setError(error)
+
+      throw error
+    } finally {
+      executing.value = false
+      executed.value = true
+    } 
+  }
+
+  async function safeExecute(): Promise<void> {
+    try {
+      await execute()
+    } catch(error) {
+      log(error)
     }
-    
-    lastExecuted.value = Date.now()
-    executing.value = false
-    
-    setNextExecution()
   }
 
   function setResponse(value: Awaited<ReturnType<TAction>>): void {
+    error.value = undefined
+    errored.value = false
```

Comment:
Kinda don't like how these are hidden within setResponse rather than being part of execute.

Reply:
Maybe but I love how simple the execute reads now